ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -631,6 +632,12 @@ List<ExchangeMetadataResponse> exchangeMetadata(ConnectedChannel connectedChanne
       DataNodeId remoteNode = replicasToReplicatePerNode.get(0).getReplicaId().getDataNodeId();
       ReplicaMetadataResponse response =
           getReplicaMetadataResponse(replicasToReplicatePerNode, connectedChannel, remoteNode);
+      if (response.getError() != ServerErrorCode.No_Error) {
```

Comment:
nit: looks like we are already checking for this error in `getReplicaMetadataResponse`.

Reply:
that's right, the blocking channel doesn't have this issue.